### PR TITLE
Worker

### PR DESCRIPTION
**Pull request by cnevinc**
related https://github.com/mozilla-tw/FirefoxLite/issues/4659

### CODE REVIEW

**Comment by cnevinc** [comment]
The new revision will not stop the cp
see https://dev.to/viniciusccarvalho/graceful-shutdown-of-ktor-applications-1h53

**Comment by cnevinc** [comment]
Hi @wangchunyang 
I'll merge this first for the UAT release this week.
Please give us a post-code-review after this PR is merged. Thank you!

**Comment by wangchunyang** [comment]
> Hi @wangchunyang
> I'll merge this first for the UAT release this week.
> Please give us a post-code-review after this PR is merged. Thank you!

Sure, go ahead please.

File: src/org/mozilla/app/push/util/SQLTransaction.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            this.commit()
+
+            this.autoCommit = true
+
+            return@use result
+        }
+    } catch (e: SQLException) {
+        e.printStackTrace()
+        logger().error("[worker][$logging][error]$e")
+        return null
```
Comment (wangchunyang) [inline src/org/mozilla/app/push/util/SQLTransaction.kt:23]: 1. exception was already printed in log. Maybe you can remove this line ` e.printStackTrace()`?
2. Might think over about `return null` or throwing a runtime exception.

**Review by wangchunyang — COMMENTED** [review]
Overall the code is very simple and clean. Great!

File: src/org/mozilla/app/push/util/SQLTransaction.kt (hunk 1 above)
Comment (cnevinc) [reply src/org/mozilla/app/push/util/SQLTransaction.kt:23]: Can I just rethrow the exception here?

**Pull request merged**